lib/shopify-cli/js_system.rb

```diff
@@ -0,0 +1,95 @@
+require 'shopify_cli'
+
+module ShopifyCli
+  ##
+  # ShopifyCli::JsSystem allows conditional system calls of npm or yarn commands.
+  #
+  class JsSystem
+    include SmartProperties
+
+    YARN_CORE_COMMAND = 'yarn'
+    NPM_CORE_COMMAND = 'npm'
+
+    class << self
+      ##
+      # Proxy to instance method ShopifyCli::JsSystem.new.yarn?.
+      #
+      # #### Parameters
+      # - `ctx`: running context from your command
```

Comment:
should this be "for" ? (I'm not sure here)

Reply:
`context` comes from the `Command` and this is the wording used in other places of the documentation. (This was taken from the JsDeps rubydocs).